webui/react/src/components/TaskList.tsx

```diff
@@ -552,7 +558,13 @@ const TaskList: React.FC<Props> = ({ workspaceId }: Props) => {
   );
 
   const JupyterLabButton = () => {
-    return <Button onClick={() => openJupyterLabModal()}>Launch JupyterLab</Button>;
+    return (
+      <Button
+        disabled={workspace ? !canCreateWorkspaceNSC(workspace) : !canCreateNSC}
```

Comment:
you're probably on this already but I had to change this to `{ workspace }` to run